utilities/transactions/transaction_base.h

```diff
@@ -330,8 +330,14 @@ class TransactionBaseImpl : public Transaction {
   // Optimistic Transactions will keep note the requested locks (not actually
   // locked), and do conflict checking until commit time based on the tracked
   // lock requests.
+  //
+  // Declared as public because
+  //  - RangeLocking's lock escalation may replace it
+  //  - RangeTreeLockManager::UnLock needs access to it
```

Comment:
can we have a public getter method instead of exposing this member directly as public?

Reply:
Yes. Done.